### DISCUSSION TITLE
Nested ReactFlow Component Causes Node Position Offset

### DISCUSSION THREAD

**Discussion by iGmainC** [xyflow/xyflow#4743] [Q&A]
I'm encountering an issue with ReactFlow when trying to nest a ReactFlow component inside a custom node. The nodes within the nested ReactFlow component are experiencing a position offset.

Here's a brief description of my setup:

- I have a custom node that contains a nested ReactFlow component.
- The canvas dragging and node dragging functionalities within the nested ReactFlow component are not working.
- The nodes inside the nested ReactFlow component are not positioned correctly; they appear to be offset from their intended positions.

Below is a snippet of my code for reference:

```ts
const LoopNode = React.memo(({ id, data }: {
  id: string;
  data: { name: string, desc: string, schema: any };
}) => {
  return (
    <Card className='w-min'>
      <CardHeader className='pb-0 drag-handle w-full'>{data.name}</CardHeader>
      <CardBody className='flex flex-row gap-1'>
        <div className='w-[18rem] h-[12rem] bg-slate-400 z-20'></div>
        <div className='w-[30rem] h-[12rem] rounded-lg z-10'
        style={{
          boxShadow: '0 0 6px 4px rgba(0,0,0,0.1), 0 2px 4px rgba(0,0,0,0.1)'
        }}
        >
          <ReactFlowProvider>
            <ReactFlow proOptions={{
              hideAttribution: true,
            }}
            nodes={[{
              id: '1',
              type: 'startNode',
              data: { name: 'Test Node' },
              position: { x: 150, y: 25 },
            }]}
            >
              <Background variant={BackgroundVariant.Dots} gap={24} />
            </ReactFlow>
          </ReactFlowProvider>
        </div>
      </CardBody>
    </Card>
  );
});
```

I have also attached a video demonstrating the issue. Any help or suggestions on how to resolve this would be greatly appreciated!

https://github.com/user-attachments/assets/9736e018-ac66-41e1-a903-8f8b386c435c

PS: I noticed that when the zoom level of the outer ReactFlow component is set to 1 (no zoom), the nodes inside the nested ReactFlow component do not experience any position offset.

**Comment by PythonLawrence** [comment]
@iGmainC  I have had extremely similar issues before without nested components, but correcting the coordinate transform always fixed it. Did you resolve this or did the nesting of components cause an issue you couldn't circumvent??

**Reply by bcakmakoglu** [reply]
Nested Flows aren't really supported because of the outer vs inner transform issue afaik
You could just render it in a separate space that's not nested or in something like a Modal though.
